### PR TITLE
feat(core): support creating cloud workspaces to different servers

### PR DESCRIPTION
**Pull request by JimmFly**

### CODE REVIEW

**Comment by JimmFly** [comment]
* **#9006** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9006?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9006?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/core/src/modules/dialogs/constant.ts [hunk 1]
```diff
@@ -31,9 +43,13 @@ export type GLOBAL_DIALOG_SCHEMA = {
     workspaceMetadata?: WorkspaceMetadata | null;
     scrollAnchor?: string;
   }) => void;
-  'sign-in': (props: { server?: string; step?: 'sign-in' }) => void;
+  'sign-in': (props: { server?: string; step?: string }) => void;
   'change-password': (props: { server?: string }) => void;
   'verify-email': (props: { server?: string; changeEmail?: boolean }) => void;
+  'enable-cloud': (props: {
+    workspace?: Workspace;
```
Comment (EYHN) [inline packages/frontend/core/src/modules/dialogs/constant.ts]: use workspace id here

File: packages/frontend/core/src/modules/dialogs/constant.ts [hunk 2]
```diff
@@ -31,9 +43,13 @@ export type GLOBAL_DIALOG_SCHEMA = {
...
+    options?: ConfirmEnableCloudOptions;
+  }) => void;
```
Comment (EYHN) [inline packages/frontend/core/src/modules/dialogs/constant.ts]:
```suggestion
  }) => boolean;
```
just return boolean to indicate whether the operation was completed successfully

File: packages/frontend/core/src/modules/dialogs/constant.ts [hunk 3]
```diff
@@ -31,9 +43,13 @@ export type GLOBAL_DIALOG_SCHEMA = {
     workspaceMetadata?: WorkspaceMetadata | null;
     scrollAnchor?: string;
   }) => void;
-  'sign-in': (props: { server?: string; step?: 'sign-in' }) => void;
+  'sign-in': (props: { server?: string; step?: string }) => void;
   'change-password': (props: { server?: string }) => void;
   'verify-email': (props: { server?: string; changeEmail?: boolean }) => void;
+  'enable-cloud': (props: {
+    workspace?: Workspace;
+    options?: ConfirmEnableCloudOptions;
```
Comment (EYHN) [inline packages/frontend/core/src/modules/dialogs/constant.ts]: There is no need to add a new type, just list it here

```
openPageId?: string;
serverId?: string;
```

File: packages/common/infra/src/modules/workspace/services/transform.ts [hunk 4]
```diff
@@ -23,14 +23,15 @@ export class WorkspaceTransformService extends Service {
...
+    accountId: string,
+    flavour: string
   ): Promise<WorkspaceMetadata> => {
     assertEquals(local.flavour, 'local');
 
     const localDocStorage = local.engine.doc.storage.behavior;
 
     const newMetadata = await this.factory.create(
-      'affine-cloud',
+      flavour ?? 'affine-cloud',
```
Comment (EYHN) [inline packages/common/infra/src/modules/workspace/services/transform.ts]: flavour will never be null or undefined?

File: packages/frontend/core/src/components/hooks/affine/use-enable-cloud.tsx [hunk 5]
```diff
@@ -42,7 +46,8 @@ export const useEnableCloud = () => {
         if (!account) return;
         const { id: newId } = await workspacesService.transformLocalToCloud(
           ws,
-          account.id
+          account.id,
+          'affine-cloud'
```
Comment (forehalo) [inline packages/frontend/core/src/components/hooks/affine/use-enable-cloud.tsx:50]: is this right?

**Review by forehalo — APPROVED** [review]

File: packages/frontend/core/src/components/hooks/affine/use-enable-cloud.tsx (hunk 5 above)
Comment (JimmFly) [reply packages/frontend/core/src/components/hooks/affine/use-enable-cloud.tsx:50]: If no additional servers are added, should there only be affine-cloud? 
Currently, the workspace flavor only has local, affine-cloud, and "serverID".
If the default server base url is modified, it should still be affine-cloud by default, right?🧐

**Pull request merged**